warp/handlers/signature_request_p2p.go

```diff
@@ -0,0 +1,146 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package handlers
+
+import (
+	"context"
+	"time"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p"
+	"github.com/ava-labs/avalanchego/proto/pb/sdk"
+	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/subnet-evm/warp"
+	"google.golang.org/protobuf/proto"
+)
+
+var _ p2p.Handler = (*SignatureRequestHandlerP2P)(nil)
+
+const (
+	ErrFailedToParse = iota
+	ErrFailedToGetSig
+	ErrFailedToMarshal
+)
+
+// SignatureRequestHandlerP2P serves warp signature requests using the p2p
+// framework from avalanchego. It is a peer.RequestHandler for
+// message.MessageSignatureRequest.
+type SignatureRequestHandlerP2P struct {
+	backend warp.Backend
+	codec   codec.Manager
+	stats   *handlerStats
+}
+
+func NewSignatureRequestHandlerP2P(backend warp.Backend, codec codec.Manager) *SignatureRequestHandlerP2P {
```

Comment:
We can probably remove the existing handler, unless we have some concerns about it being in use.

Reply:
I think we need to have this implemented in relayer as well before removing from here? cc @michaelkaplan13 

I'd say we should keep it around until E-Upgrade.

Reply:
Yeah the AWM relayer uses the existing handlers today, so I think we should leave them in place until after the E-upgrade goes live and everyone has a chance to switch. They can probably be removed soon after that though 👍

Reply:
Sounds good. I can add a comment about this.